dissect/target/plugins/apps/webservers/nginx.py

```diff
@@ -0,0 +1,71 @@
+import re
+from datetime import datetime
+from pathlib import Path
+from typing import Iterator
+
+from dissect.target import plugin
+from dissect.target.helpers.fsutil import open_decompress
+from dissect.target.plugins.apps.webservers.webservers import WebserverRecord
+from dissect.target.target import Target
+
+LOG_REGEX = re.compile(
+    r'(?P<remote_ip>.*?) - (?P<remote_user>.*?) \[(?P<datetime>\d{2}\/[A-Za-z]{3}\/\d{4}:\d{2}:\d{2}:\d{2} (\+|\-)\d{4})\] "(?P<url>.*?)" (?P<status_code>\d{3}) (?P<bytes_sent>\d+) (["](?P<referer>(\-)|(.+))["]) "(?P<useragent>.*?)"',  # noqa: E501
+    re.IGNORECASE,
+)
+
+
+class NginxPlugin(plugin.Plugin):
+    __namespace__ = "nginx"
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self.log_paths = self.get_log_paths()
+
+    @plugin.internal
+    def get_log_paths(self) -> list[Path]:
+        log_paths = []
+        default_logs_dir = self.target.fs.path("/var/log/nginx")
+        if default_logs_dir.exists():
+            log_paths = [self.target.fs.path(p) for p in default_logs_dir.glob("access.log*")]
+
+        # Check for custom paths in nginx install config
+        if (config_file := self.target.fs.path("/etc/nginx/nginx.conf")).exists():
+            for line in config_file.open("rt"):
+                line = line.strip()
+                if "access_log " in line:
+                    p = list(filter(None, line.strip().split(" ")))[1]
+                    if (path := self.target.fs.path(p).parent).exists():
```

Comment:
Won't accessing the `.parent` here give you the directory where the log file is located? There's no iterating over any files happening after this.

Maybe a good idea to change `get_log_paths` to instead be `get_log_dirs` and do the glob for `access.log*` in the parsing function below. Otherwise you're doing the same glob in multiple places.

Same comment for the Caddy plugin by the way.

Reply:
Good catch. Improved this behaviour in [01412a9](https://github.com/fox-it/dissect.target/pull/127/commits/01412a927a93b4751bd6abaae3cbba0bcc66802c).